src/mgmt/srv_drpc.c

```diff
@@ -1899,6 +1899,85 @@ ds_mgmt_drpc_dev_replace(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	D_FREE(resp);
 }
 
+void
+ds_mgmt_drpc_dev_identify(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	struct drpc_alloc	alloc = PROTO_ALLOCATOR_INIT(alloc);
+	Mgmt__DevIdentifyReq	*req = NULL;
+	Mgmt__DevIdentifyResp	*resp = NULL;
+	uint8_t			*body;
+	size_t			 len;
+	uuid_t			 dev_uuid;
+	int			 rc = 0;
+
+	/* Unpack the inner request from the drpc call body */
+	req = mgmt__dev_identify_req__unpack(&alloc.alloc,
+					     drpc_req->body.len,
+					     drpc_req->body.data);
+
+	if (alloc.oom || req == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		D_ERROR("Failed to unpack req (dev identify)\n");
+		return;
+	}
+
+	if (strlen(req->dev_uuid) == 0) {
+		D_ERROR("Device UUID for identify command is empty\n");
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		mgmt__dev_identify_req__free_unpacked(req, &alloc.alloc);
+		return;
+	}
+
+	D_INFO("Received request to identify device %s\n", req->dev_uuid);
+
+	D_ALLOC_PTR(resp);
+	if (resp == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		D_ERROR("Failed to allocate daos response ref\n");
+		mgmt__dev_identify_req__free_unpacked(req, &alloc.alloc);
+		return;
+	}
+
+	/* Response status is populated with SUCCESS on init. */
+	mgmt__dev_identify_resp__init(resp);
```

Comment:
immediately set the "empty string" to NULL, otherwise, error cleanup code will try to free the "empty string".